Feat file upload

oussamadhouib opened the pull request:
This pull request introduces file upload functionality to the chat system

miurla commented on app/api/upload/route.ts:
```diff
@@ -0,0 +1,83 @@
+  const sanitizedFileName = sanitizeFilename(file.name)
+  const filePath = `${Date.now()}-${sanitizedFileName}`
+  const supabase = await createClient()
+  const { error } = await supabase.storage
```
Please update the file path to include a hierarchical structure for better file management. This is a requirement for proper user data management and future maintenance.

Required change:
```typescript
const filePath = `${userId}/chats/${chatId}/${Date.now()}-${sanitizedFileName}`
```

This structure will ensure files are properly organized by user and chat, making cleanup operations easier when users are deleted and allowing better organization of uploaded files by conversation context.

miurla commented on components/chat.tsx:
```diff
@@ -274,11 +277,39 @@ export function Chat({
     e.preventDefault()
-    handleSubmit(e)
+
+    const uploadedAttachments = await Promise.all(
```
Please change the file upload timing. Currently, files are uploaded when the message is sent, which makes users wait for uploads to complete before their message is processed.\n\nInstead, implement immediate upload after file selection:\n1. Upload files immediately after they are selected (in the file selection handler)\n2. Show upload progress to users\n3. Store the uploaded file information (URLs) to use when the message is sent\n4. Only include successfully uploaded files in the message\n\nThis change will significantly improve user experience by:\n- Reducing perceived latency when sending messages\n- Allowing parallel uploads while users are typing\n- Providing clear feedback on upload status before message submission\n\nYou can reference the Vercel AI Chatbot implementation at https://github.com/vercel/ai-chatbot/blob/main/components/multimodal-input.tsx which uses this pattern. In their implementation, `handleFileChange` handles the upload process immediately after file selection rather than waiting for form submission."

miurla replied:
Please change the file upload timing. Currently, files are uploaded when the message is sent, which makes users wait for uploads to complete before their message is processed.\n\nInstead, implement immediate upload after file selection:\n\n1. Upload files immediately after they are selected (in the file selection handler)\n2. Show upload progress to users\n3. Store the uploaded file information (URLs) to use when the message is sent\n4. Only include successfully uploaded files in the message\n\nThis change will significantly improve user experience by:\n\n- Reducing perceived latency when sending messages\n- Allowing parallel uploads while users are typing\n- Providing clear feedback on upload status before message submission\n\nYou can reference the Vercel AI Chatbot implementation at https://github.com/vercel/ai-chatbot/blob/main/components/multimodal-input.tsx which uses this pattern. In their implementation, `handleFileChange` handles the upload process immediately after file selection rather than waiting for form submission."

miurla commented:
Thank you for this great PR! The file upload implementation looks solid and will be a valuable addition to the application.

I've left two specific comments on the code that I'd appreciate if you could review:
1. Adjusting the file path structure to include user and chat IDs for better organization
2. Changing the upload timing to happen immediately after file selection rather than at message submission time

After merging, I'll make some minor adjustments to the user message styles and layout to ensure consistent UI throughout the application. 

Great work on this feature! 👍

oussamadhouib commented:
@miurla thank you for the thoughtful feedback
I've addressed the two points you mentioned:
- File path structure: Updated to include userId and chatId for improved organization and future scalability.
- Upload timing: Adjusted so files are now uploaded immediately upon selection, with upload progress and status handled before message submission

miurla commented:
Thank you for the quick fixes and your great contribution! I've tested the functionality and everything works perfectly. Your implementation of file uploads adds a valuable feature to the application, and I appreciate how you addressed all the feedback points promptly.
